### DISCUSSION TITLE
Use contextvars.ContextVar to pass request object through the view to the serializer.context

### DISCUSSION THREAD

**Discussion by daler-sz** [encode/django-rest-framework#8904] [Ideas & Suggestions]
We often use APIViews and override generic methods in combinatoin with serializers. And there is often need to manually pass request object to serializer.context from APIViews and generic methods. So, such code does not look good and may lead to bugs, since we may forget to pass the request object to context. 

I think it is good idea to keep request object in context (using [contextvars](https://docs.python.org/3/library/contextvars.html) module from standart library of python) and get this request object from contextvar. 

So, if we create serializer object in view, this serializer object would always have request object in his context even if we don't pass it.

**Comment by auvipy** [comment]
can you please share code example of what you are actually doing and want to achieve?

**Comment by daler-sz** [comment]
```python
class SomeView(APIView):
    def post(self, request)
        serializer = InputSerializer(data=request.data, context={'request': request})
        ....
        ....
        output_data = OutputSerializer(data=..., context={'request': request}).data
```
I want to get a rid of ```context={'request': request}``` everywhere so the request object pass to serializers context automatically (if it is not provided manually). As I said, it is quite possible with contextvars.ContextVar

**Comment by daler-sz** [comment]
@auvipy so, basically, what do you think about this idea? I can provide pull request with it, if needed. But it would only work in python3.7 (and higher versions), since contextvars were introduced in this version

**Reply by auvipy** [reply]
you can start a draft implementation. we will remove 3.6 in few months. as this will change the API surface, so we would need a consensus on the design decision

**Reply by daler-sz** [reply]
I am almost sure this wouldn't change API surface at all. Thank you for feedback
